saleor_app_framework/app_framework/core/conf.py

```diff
@@ -1,27 +1,59 @@
 import importlib
+import logging
 import os
-
-SETTINGS_ENV_VARIABLE = "APP_SETTINGS_MODULE"
-
-
-class ImproperlyConfigured(Exception):
-    pass
-
-
-class Settings(object):
-    def __init__(self, settings_module):
-        # store the settings module in case someone later cares
-        self.SETTINGS_MODULE = settings_module
-
-        mod = importlib.import_module(self.SETTINGS_MODULE)
-
-        self._explicit_settings = set()
-        for setting in dir(mod):
-            if setting.isupper():
-                setting_value = getattr(mod, setting)
-
-                setattr(self, setting, setting_value)
-                self._explicit_settings.add(setting)
-
-
-settings = Settings(os.environ.get(SETTINGS_ENV_VARIABLE))
+from functools import lru_cache
+from typing import Optional
+
+from pydantic import BaseSettings
+
+SETTINGS_ENV_VARIABLE = "APP_SETTINGS"
+
+
+logger = logging.getLogger(__file__)
+PROJECT_DIR = os.path.dirname(__file__)
+
+
+class Settings(BaseSettings):
+    app_name: str = "App"
+    project_dir: str
+    static_dir: str
+    templates_dir: str
+    manifest_path: str
+    debug: bool = True
+    dev_saleor_domain: Optional[str] = None
+    dev_saleor_token: Optional[str] = None
+
+
+def import_string(dotted_path):
+    """
+    Import a dotted module path and return the attribute/class designated by the
+    last name in the path. Raise ImportError if the import failed.
+    """
+    try:
+        module_path, class_name = dotted_path.rsplit(".", 1)
+    except ValueError as err:
+        raise ImportError("%s doesn't look like a module path" % dotted_path) from err
+
+    module = importlib.import_module(module_path)
+
+    try:
+        return getattr(module, class_name)
+    except AttributeError as err:
+        raise ImportError(
+            'Module "%s" does not define a "%s" attribute/class'
+            % (module_path, class_name)
+        ) from err
+
+
+@lru_cache()
+def get_settings() -> Settings:
+    settings_path = os.environ.get(SETTINGS_ENV_VARIABLE)
+    if not settings_path:
+        raise Exception(
+            f"Env {SETTINGS_ENV_VARIABLE} was not provided. Provide python path to "
+            f"project's settings class."
+        )
+    settings = import_string(settings_path)()
```

Comment:
Assuming `APP_SETTINGS` env points to an instance of `Settings` this would not be callable. You want to make an instance of the settings like `app_settings = Settings(...)` and use the `app_settings` instance in the service, as all the evaluation (i.e. env var evaluation) from `pydantic.BaseSettings` happens on init.